cfme/configure/settings.py

```diff
@@ -53,272 +49,280 @@ def __init__(self, description=None, scope=None, days=None, hours=None, timezone
             end = "saved" if self.appliance.version > '5.7' else "added"
-            flash.assert_success_message('Time Profile "{}" was {}'
-                                         .format(self.description, end))
+            view.flash.assert_message('Time Profile "{}" was {}'.format(self.description, end))
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(self.timeprofile_form, {'description': updates.get('description'),
-                                     'scope': updates.get('scope'),
-                                     'timezone': updates.get('timezone')},
-             action={version.LOWEST: form_buttons.save,
-                     '5.7': self.save_edit_button})
-        flash.assert_success_message(
-            'Time Profile "{}" was saved'.format(updates.get('description', self.description)))
+        view = navigate_to(self, 'All')
+        rows = view.timeprofile_form.table
+        for row in rows:
+            if row.Description.text == self.description:
+                row[0].check()
+        view.timeprofile_form.configuration.item_select('Edit selected Time Profile')
+        changed = view.timeprofile_form.fill({
+            'description': updates.get('description'),
+            'scope': updates.get('scope'),
+            'days': updates.get('days'),
+            'hours': updates.get('hours'),
+            'timezone': updates.get('timezone'),
+        })
+        if changed:
+            view.timeprofile_form.save_edit_button.click()
+            view.flash.assert_message(
+                'Time Profile "{}" was saved'.format(updates.get('description', self.description)))
 
     def copy(self):
-        navigate_to(self, 'All')
-        row = timeprofile_table.find_row_by_cells({'description': self.description})
-        sel.check(sel.element(".//input[@type='checkbox']", root=row[0]))
-        cfg_btn('Copy selected Time Profile')
-        new_timeprofile = Timeprofile(description=self.description + "copy",
-                         scope=self.scope)
-        fill(self.timeprofile_form, {'description': new_timeprofile.description,
-                              'scope': new_timeprofile.scope},
-             action=self.save_button)
-        end = "saved" if self.appliance.version > '5.7' else "added"
-        flash.assert_success_message('Time Profile "{}" was {}'
-                                     .format(new_timeprofile.description, end))
+        view = navigate_to(self, 'All')
+        rows = view.timeprofile_form.table
+        for row in rows:
+            if row.Description.text == self.description:
+                row[0].check()
+        view.timeprofile_form.configuration.item_select('Copy selected Time Profile')
+        new_timeprofile = Timeprofile(description="{} copy".format(self.description),
+                                      scope=self.scope)
+        changed = view.timeprofile_form.fill({
+            'description': "{} copy".format(self.description),
```

Comment:
REQUIRED: You want to pass the new name as a parameter to copy, while you are here. You can put it as optional with the fallback to current behaviour of autogenerating the copied name.